### DISCUSSION TITLE
File/Image Upload fails

### DISCUSSION THREAD

**Discussion by BanditsBacon** [toeverything/AFFiNE#6045] [Q&A] [answered]
I use the selfhosted Version on external Host-Machines with the Standard docker-compose.yml. 
https://github.com/toeverything/AFFiNE/blob/stable/.github/deployment/self-host/compose.yaml
Only with different Paths for the Volumes on the Host Machine 

1. Upload a Image with /image
2. It failed with 
<img width="586" alt="Bildschirmfoto 2024-03-07 um 11 42 31" src="https://github.com/toeverything/AFFiNE/assets/2463467/0c829dd3-aaf6-4201-8b79-578e496f1e49">

and console log.
<img width="1272" alt="Bildschirmfoto 2024-03-07 um 11 51 35" src="https://github.com/toeverything/AFFiNE/assets/2463467/bc934d11-2cb9-48f1-bbb8-3b7a8c68c9e9">

If i run it locally with Docker and localhost:3010 then it works
I'm not sure if's a bug.

**Comment by fourdim** [comment, marked as answer]
It's because you are not serving self-host AFFiNE Cloud under HTTPS (Secure Context).
https://developer.mozilla.org/en-US/docs/Web/API/Crypto/subtle
https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts
Many features of browsers are disabled.
CC @EYHN

**Comment by BanditsBacon** [comment]
For me it was solved in this Issue https://github.com/toeverything/AFFiNE/issues/6049
